packages/automerge-repo/src/storage/StorageSubsystem.ts

```diff
@@ -1,97 +1,80 @@
-import * as Automerge from "@automerge/automerge"
-import { DocumentId } from "../DocHandle"
+import * as A from "@automerge/automerge"
+import { DocumentId } from "../types"
+import { mergeArrays } from "../helpers/mergeArrays"
+import { StorageAdapter } from "./StorageAdapter"
 
-export interface StorageAdapter {
-  load(docId: string): Promise<Uint8Array | null>
-  save(docId: string, data: Uint8Array): void
-  remove(docId: string): void
-}
-
-function mergeArrays(myArrays: Uint8Array[]) {
-  // Get the total length of all arrays.
-  let length = 0
-  myArrays.forEach((item) => {
-    length += item.length
-  })
-
-  // Create a new array with total length and merge all source arrays.
-  const mergedArray = new Uint8Array(length)
-  let offset = 0
-  myArrays.forEach((item) => {
-    mergedArray.set(item, offset)
-    offset += item.length
-  })
-
-  return mergedArray
-}
 export class StorageSubsystem {
-  storageAdapter: StorageAdapter
+  #storageAdapter: StorageAdapter
+  #changeCount: Record<DocumentId, number> = {}
```

Comment:
yeah this approach is poor and should be replaced by something hash-based for better concurrency (later)